### ISSUE TITLE
UniTask.Delay results in "Pending" status forever

### ISSUE THREAD

**Issue by Nyankoo** [Cysharp/UniTask#58]
I'm trying to use "await UniTask.Delay", but every time that line gets executed, the UniTask Tracker lists two new entries with status pending ("UniTask.DelayPromise" and "UniTask.CompletionSource").

```
private void startMethod()
    {
        StartCoroutine(UniTask.ToCoroutine(asyncMethod));
    }

private async UniTask asyncMethod()
    {
        //do something
        await UniTask.Delay(5000);
        //do something
    }
```

**Comment by Nyankoo** [comment]
Follow-up: it seems that this is an issue with Unity 2019.3.0f6. The above code works in Unity 2019.2.19f1.

Anyone also encountered this issue?

**Comment by neuecc** [comment]
I can't reproduce it in 2019.3.9f1.

```csharp
void Start()
{
    startMethod();
}

private void startMethod()
{
    StartCoroutine(UniTask.ToCoroutine(asyncMethod));
}

private async UniTask asyncMethod()
{
    UnityEngine.Debug.Log("async start");

    await UniTask.Delay(5000);

    UnityEngine.Debug.Log("async completed");
}
```

![image](https://user-images.githubusercontent.com/46207/79430913-9e2a0580-8004-11ea-9790-499f726f3e38.png)

**Comment by Nyankoo** [comment]
I'm doing something similar:

```
void Start()
{
    startMethod();
}

private void startMethod()
{
    StartCoroutine(UniTask.ToCoroutine(asyncMethod));
}

private async UniTask asyncMethod()
{
        while(!AccountManager.Instance.IsInitialized)
        {
            Debug.Log(AccountManager.Instance.IsInitialized);

            await UniTask.Delay(System.TimeSpan.FromSeconds(1));

            Debug.Log(AccountManager.Instance.IsInitialized);
        }
}
```

`AccountManager.Instance` is a Singleton/Monobehavior that gets created in the first `AccountManager.Instance.IsInitialized` call. The first debug line prints "false", as the `Start()` method of theMonoBehavior still needs to complete. After that, the Delay call happens, but after 1 second (or more), the second Debug line is never called and `UniTask.DelayPromise` with status Pending is visible in the UniTask Tracker.

**Comment by neuecc** [comment]
Do you use any other packages?
UniTask is customizing PlayerLoop.
However, it is possible that other packages have overridden them.

**Comment by Nyankoo** [comment]
@neuecc Do you mean packages installed via UPM or other assets/plugins?

**Comment by neuecc** [comment]
for example, ECS package override Playerloop.
In this case, re-initialization will solve the problem.

```
var playerLoop = Unity.Entities.ScriptBehaviourUpdateOrder.CurrentPlayerLoop;
PlayerLoopHelper.Initialize(ref playerLoop);
```

**Comment by Nyankoo** [comment]
@neuecc I'm not using ECS in my project.

**Comment by phobos2077** [comment]
If GetCurrentPlayerLoop() is not used in UniTask, then it should be updated to use it. It's the only way to properly modify PlayerLoop w/o breaking any other code that is updating PlayerLoop as well. We use ECS as well and having to use workaround for something that should be taken care of by UniTask itself is weird.

**Comment by neuecc** [comment]
UniTask already uses GetCurrentPlayerLoop but ECS does not(uses GetDefaultPlayerLoop).
You should complain to ECS.

**Comment by phobos2077** [comment]
Thanks for clarification. This issues should hopefully be resolved in ECS 0.13.0 as they've changed the way worlds are added to player loop.
